### PR TITLE
Snowflake: Fix Alter Warehouse

### PR DESCRIPTION
**Pull request by AidanHarveyNelson**
### Brief summary of the change made
Added optional parameter on the Snowflake Alter Warehouse SET command step.

While the documentation lists that field as required by first running `USE WAREHOUSE` or selecting a Warehouse prior to running the command it will target the active selected warehouse.

https://github.com/sqlfluff/sqlfluff/issues/4342

### Pull Request checklist
- [ X ] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by greg-finley** [comment]
Thanks! Next time, please write something like closes #4342 so they are linked together and the issue auto-closes when this merges.

File: test/fixtures/dialects/snowflake/snowflake_alter_warehouse.sql [hunk 1]
```diff
@@ -21,3 +21,7 @@ alter warehouse LOAD_WH SET MAX_CONCURRENCY_LEVEL = 1;
 alter warehouse LOAD_WH UNSET STATEMENT_QUEUED_TIMEOUT_IN_SECONDS;
 alter warehouse LOAD_WH UNSET WAREHOUSE_SIZE;
 alter warehouse LOAD_WH UNSET WAREHOUSE_SIZE, WAIT_FOR_COMPLETION;
+
+-- Real Life Use Cases Taken from Issue 4342 https://github.com/sqlfluff/sqlfluff/issues/4342
```
Comment (greg-finley) [inline test/fixtures/dialects/snowflake/snowflake_alter_warehouse.sql:25]:
```suggestion
```

Unnecessary to give this much info. We could do a `git blame` if we really wanted to know why this was added (and it's not controversial).

File: test/fixtures/dialects/snowflake/snowflake_alter_warehouse.sql [hunk 2]
```diff
@@ -21,3 +21,6 @@ alter warehouse LOAD_WH SET MAX_CONCURRENCY_LEVEL = 1;
 alter warehouse LOAD_WH UNSET STATEMENT_QUEUED_TIMEOUT_IN_SECONDS;
 alter warehouse LOAD_WH UNSET WAREHOUSE_SIZE;
 alter warehouse LOAD_WH UNSET WAREHOUSE_SIZE, WAIT_FOR_COMPLETION;
+
+ALTER WAREHOUSE SET WAREHOUSE_SIZE='X-LARGE';
+alter warehouse set warehouse_size=medium
```
Comment (greg-finley) [inline test/fixtures/dialects/snowflake/snowflake_alter_warehouse.sql:26]: I would usually add the ending semicolon but will just leave it for now so we don't have to rehash.

**Comment by greg-finley** [comment]
Also check the title change, usually best to write "<Dialect>: <some short description of the change>"

**Review by greg-finley — APPROVED** [review]
Thanks for the contribution! 🙌

**Comment by AidanHarveyNelson** [comment]
I appreciate the advice, first commit to this project and open source projects so still learning the expectations

**Pull request merged**